hkmc2/shared/src/main/scala/hkmc2/codegen/HandlerLowering.scala

```diff
@@ -49,29 +56,28 @@ import HandlerLowering.*
 
 class HandlerLowering(using TL, Raise, Elaborator.State, Elaborator.Ctx):
 
-  private def funcLikeHandlerCtx(ctorThis: Option[Path]) =
-    HandlerCtx(true, false, ctorThis, state =>
-      val tmp = freshTmp()
+  private def funcLikeHandlerCtx(ctorThis: Option[Path], isHandlerMtd: Bool, nme: Str) =
+    HandlerCtx(!isHandlerMtd, false, isHandlerMtd, false, nme, ctorThis, state =>
       blockBuilder
-        .assignFieldN(state.res.tail, nextIdent, Instantiate(
+        .assignFieldN(state.res.contTrace.last, nextIdent, Instantiate(
           state.cls.selN(Tree.Ident("class")),
           Value.Lit(Tree.IntLit(state.uid)) :: Nil))
-        .assignFieldN(state.res, tailIdent, state.res.tail.next)
+        .assignFieldN(state.res.contTrace, lastIdent, state.res.contTrace.last.next)
         .ret(state.res))
-  private val functionHandlerCtx = funcLikeHandlerCtx(N)
-  private val topLevelCtx = HandlerCtx(true, true, N, _ => rtThrowMsg("Unhandled effects"))
-  private def ctorCtx(ctorThis: Path) = funcLikeHandlerCtx(S(ctorThis))
+  private def functionHandlerCtx(nme: Str) = funcLikeHandlerCtx(N, false, nme)
+  private def topLevelCtx(nme: Str) = HandlerCtx(false, true, false, false, nme, N, _ => rtThrowMsg("Unhandled effects"))
+  private def ctorCtx(ctorThis: Path, nme: Str) = funcLikeHandlerCtx(S(ctorThis), false, nme)
+  private def handlerMtdCtx(nme: Str) = funcLikeHandlerCtx(N, true, nme)
   private def handlerCtx(using HandlerCtx): HandlerCtx = summon
-  private val predefPath: Path = State.globalThisSymbol.asPath.selN(Tree.Ident("Predef"))
-  private val predefSym: ModuleSymbol = ctx.builtins.Predef
-  private val effectSigPath: Path = predefPath.selN(Tree.Ident("__EffectSig")).selN(Tree.Ident("class"))
-  private val effectSigSym: ClassSymbol = predefSym.tree.definedSymbols.get("__EffectSig").get.asCls.get
-  private val contClsPath: Path = predefPath.selN(Tree.Ident("__Cont")).selN(Tree.Ident("class"))
-  private val retClsPath: Path = predefPath.selN(Tree.Ident("__Return")).selN(Tree.Ident("class"))
-  private val retClsSym: ClassSymbol = predefSym.tree.definedSymbols.get("__Return").get.asCls.get
-  private val appendInContPath: Path = predefPath.selN(Tree.Ident("__appendInCont"))
-  private val mkEffectPath: Path = predefPath.selN(Tree.Ident("__mkEffect"))
-  private val handleBlockImplPath: Path = predefPath.selN(Tree.Ident("__handleBlockImpl"))
+  private val runtimePath: Path = State.runtimeSymbol.asPath
+  private val runtimeSym: ModuleSymbol = ctx.builtins.Runtime
+  private val effectSigPath: Path = runtimePath.selN(Tree.Ident("EffectSig")).selN(Tree.Ident("class"))
+  private val effectSigSym: ClassSymbol = runtimeSym.tree.definedSymbols.get("EffectSig").get.asCls.get
```

Comment:
For now, just create a symbol from scratch representing `EffectSig` (you can add this symbol to `Elaborator.State`).

I think it doesn't matter that this won't be exactly the same symbol as the one we process when compiling `Runtime.mls`.